flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/VertexFinishedStateCheckerTest.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+import org.apache.flink.runtime.OperatorIDPair;
+import org.apache.flink.runtime.executiongraph.ExecutionGraph;
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.DistributionPattern;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.testtasks.NoOpInvokable;
+import org.apache.flink.util.FlinkRuntimeException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singletonList;
+
+/** This tests verifies the checking logic of {@link VertexFinishedStateChecker}. */
+public class VertexFinishedStateCheckerTest {
+
+    @Rule public ExpectedException thrown = ExpectedException.none();
```

Comment:
Could we rewrite it with JUnit 5 already? I believe it should be as simple as changing the imports and switching to `assertThrows`. Example for `testRestoringPartiallyFinishedChainsFails`.

```
        final FlinkRuntimeException exception = assertThrows(
                FlinkRuntimeException.class,
                finishedStateChecker::validateOperatorsFinishedState);
        
        assertThat(exception.getMessage(), equalTo("Can not restore vertex "
                + "anon("
                + jobVertexID1
                + ")"
                + " which contain mixed operator finished state: [ALL_RUNNING, FULLY_FINISHED]"));
```